### ISSUE TITLE
Overkill of PC0030 warnings

### ISSUE THREAD

**Issue by Arthurvdv** [ALCops/Analyzers#248]
### Discussed in https://github.com/ALCops/Analyzers/discussions/246

<div type='discussions-op-text'>

<sup>Originally posted by **guidorobben** May  4, 2026</sup>
I have the following piece of code.

<img width="906" height="225" alt="image" src="https://github.com/user-attachments/assets/6df9570b-efea-4f87-934b-117913bcd4b5" />

This triggers PC0030. In this case I dont want a SetLoadFields. And I also dont want to put a pragma around this. 

Currently every Get or FindFirst causes a warning/info now. More than half of this doesnt need a SetLoadFields. Is there a way to make PC0030 less strict?

0.8.0 beta3</div>

**Comment by Arthurvdv** [comment]
@guidorobben, The rule already had some logic to determine database write operations, but it seems there was a flaw that handled if-statements wrong it seems.

This should now be resolved in `v0.8.0-beta.4`, where you should have a lot less diagnostics.

**Comment by guidorobben** [comment]
I see a lot less! 

is there a way to fix this one? I have a recref where I only need the number from? I cant add this to the setloadfields because it is a property.

<img width="1795" height="619" alt="Image" src="https://github.com/user-attachments/assets/b3df2f16-c5b1-425c-bf8f-6d74bdce1b93" />

**Comment by Arthurvdv** [comment]
Great to hear the rule is behaving more as it should be :-)

I think you can use the [RecordId.TableNo()](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/methods-auto/recordid/recordid-tableno-method) which eliminate the need to retrieve the record at all.

```AL
trigger OnAction()
var
    WorkflowTableRelation: Record "Workflow - Table Relation";
begin
    WorkflowTableRelation.SetRange("Table ID", Rec."Record ID".TableNo());
    Page.Run(Page::"Workflow - Table Relations", WorkflowTableRelation);
end;
```

**Comment by guidorobben** [comment]
will check this. Good tip.

**Comment by Arthurvdv** [comment]
@guidorobben, with the release of `v0.8.0` of ALCops, this new rule is now included in the official release.

I’m going to close this issue now, but if you have any feedback or run into bugs/false positives, feel free to reopen it or create a new issue.
